x-pack/plugins/code/server/lsp/workspace_handler.ts

```diff
@@ -244,13 +244,16 @@ export class WorkspaceHandler {
     }
   }
 
-  // todo add an unit test
   private parseLocation(location: Location) {
     const uri = location.uri;
     if (uri && uri.startsWith('file://')) {
-      const workspaceUrl = new URL(`file://${this.workspacePath}`).toString();
-      if (uri.startsWith(workspaceUrl)) {
-        const relativePath = uri.substring(workspaceUrl.length + 1);
+      const locationPath = fs.realpathSync(decodeURIComponent(uri.substring('file://'.length)));
+      const workspacePath = fs.realpathSync(decodeURIComponent(this.workspacePath));
+      if (locationPath.startsWith(workspacePath)) {
+        const relativePath = path.relative(workspacePath, locationPath);
+        if (path.sep !== '/') {
```

Comment:
so this is for the sake of path on Windows?

Reply:
yeah